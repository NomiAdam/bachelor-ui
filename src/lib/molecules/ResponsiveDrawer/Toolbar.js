import React from 'react';
import PropTypes from 'prop-types';
import DrawerToggleButton from './DrawerToggleButton';
import styled from 'styled-components';
import {maxTablet, minTablet} from "../../constants/device";
import { basicTheme, darkTheme, lightTheme } from '../../constants/theme';

const StyledToolbar = styled.header`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background: ${({ primary }) => primary ? lightTheme.white : basicTheme.purple };
  height: 56px;
`;
const StyledToolbarNavigation = styled.nav`
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: center;
  padding: 0 1rem;
  position: relative;
`;
const StyledNavigationItemWrapper = styled.div`
  display: flex;
  flex-direction: row-reverse;
  width: 100%;
  ${({ hide }) => hide ? 'width: 0;' : ''}
  ul {
    list-style: none;
    margin: 0;
    padding: 0;
    display: flex;
  }
  li {
    padding: 0 .5rem;
  }
  a {
    color: ${({ primary }) => primary ? darkTheme.black : lightTheme.white };
    text-decoration: none;
  }
  @media ${maxTablet} {
	display: none;
  }
`;

const StyledLogoWrapper = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  ${({ mobileVersion }) => mobileVersion ? '' : 'justify-content: flex-start;' }
  color: ${({ primary }) => primary ? darkTheme.black : lightTheme.white };
  @media ${minTablet} {
	left: ${({ mobileVersion }) => mobileVersion ? '45%' : '-1%'};
  }
`;

const Toolbar = ({handleClick, children, logo, mobileVersion, primary}) => (
    <StyledToolbar primary={primary}>
        <StyledToolbarNavigation>
            <DrawerToggleButton
                primary={primary}
                handleClick={handleClick}
                mobileVersion={!mobileVersion}
            />
            <StyledLogoWrapper mobileVersion={mobileVersion} primary={primary}>
                {logo}
            </StyledLogoWrapper>
            <StyledNavigationItemWrapper primary={primary} hide={mobileVersion}>
                {!mobileVersion && children}
            </StyledNavigationItemWrapper>
        </StyledToolbarNavigation>
    </StyledToolbar>
);

Toolbar.propTypes = {
    /**
     * Function to handle toggle button click, will open
     */
    handleClick: PropTypes.func,
    /**
     * Logo, either node or plain text is expected
     */
    logo: PropTypes.any,
    /**
     * Whether we should use mobile version
     */
    mobileVersion: PropTypes.bool,
    /**
     * Whether to use primary theme or secondary
     */
    primary: PropTypes.bool,
};

Toolbar.defaultProps = {
    primary: false,
};

export {StyledToolbar};
export default Toolbar;
