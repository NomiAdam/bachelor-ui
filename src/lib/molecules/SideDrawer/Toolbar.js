import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import DrawerToggleButton from './BurgerButton';
import { resolveTheme, TYPES } from '../../utils/resolveTheme';

const StyledToolbar = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  background-color: ${ resolveTheme(TYPES.BACKGROUND_COLOR) };
  height: 100%;
  min-height: 100vh;
  width: ${ ({ isOpen }) => (isOpen ? '300' : '50') }px;
  overflow: hidden;
  transition: all .5s ease;
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  flex-direction: column;
`;

const Toolbar = ({
    handleClick, primary, isOpen, children, backgroundColor,
}) => (
    <StyledToolbar primary={ primary } isOpen={ isOpen } backgroundColor={ backgroundColor }>
        <DrawerToggleButton
            primary={ primary }
            handleClick={ handleClick }
        />
        { children }
    </StyledToolbar>
);

Toolbar.propTypes = {
    /**
     * Function to handle toggle button click, will open
     */
    handleClick: PropTypes.func,
    /**
     * Whether to use primary theme or secondary
     */
    primary: PropTypes.bool,
    /**
   *  Whether is open
   */
    isOpen: PropTypes.bool,
    /**
   * String definition of custom background-color property
   */
    backgroundColor: PropTypes.string,
};

Toolbar.defaultProps = {
    primary: false,
};

export { StyledToolbar };
export default Toolbar;
