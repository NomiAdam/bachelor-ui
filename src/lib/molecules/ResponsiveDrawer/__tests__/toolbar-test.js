import React from 'react';
import Toolbar, { StyledToolbar } from '../Toolbar';
import { shallow } from 'enzyme';

describe('Toolbar', () => {
    it('should be defined', () => {
        expect(Toolbar).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<Toolbar />)).toMatchSnapshot();
    });
    it('should be of type StyledToolbar', () => {
        expect(shallow(<Toolbar/>).type()).toBe(StyledToolbar);
    });
});
