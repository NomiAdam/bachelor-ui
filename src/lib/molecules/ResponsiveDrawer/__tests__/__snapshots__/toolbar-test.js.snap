// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should match the snapshot 1`] = `
<Toolbar__StyledToolbar
  primary={false}
>
  <Toolbar__StyledToolbarNavigation>
    <DrawerToggleButton
      mobileVersion={true}
      primary={false}
    />
    <Toolbar__StyledLogoWrapper
      primary={false}
    />
    <Toolbar__StyledNavigationItemWrapper
      primary={false}
    />
  </Toolbar__StyledToolbarNavigation>
</Toolbar__StyledToolbar>
`;
